sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -397,6 +397,7 @@ object QueryExecution {
       CoalesceBucketsInJoin,
       PlanDynamicPruningFilters(sparkSession),
       PlanSubqueries(sparkSession),
+      PrepareScans,
```

Comment:
The "Output Partitioning and Dynamic Filtering" section of the design doc has a bit more context.